### ISSUE TITLE
phase (phasor notation) support

### ISSUE THREAD

**Issue by 8LWXpg** [typst/typst#2264]
### Description

I would like to request the addition of phase (phasor notation) support.

For example, it could be something in math like
```typst
$3pahse(35degree)$
```
with expected output
<img width="100" alt="image" src="https://github.com/typst/typst/assets/105704427/761e4f62-9c53-4a74-9bbe-687ad9a37a38">

### Use Case

This is frequently used with electrical engineering.

**Comment by T0mstone** [comment]
I'm not sure this is common enough to be a built-in feature.

You can achieve something similar by combining `sym.angle` with `underline`:
```typst
#let phase(x) = $#box(sym.angle, inset: (right: -0.4em, bottom: -0.2em)) underline(#x)$

$3phase(35degree)$
```
![phasor](https://github.com/typst/typst/assets/39707032/73338867-e926-4da9-a347-eedad9e493d1)

Alternatively, you could resort to drawing the diagonal line yourself (with `line`) if you want it to be at a specific angle.

**Comment by 8LWXpg** [comment]
Okay, so I made one that looks close enough with pixel peeping.
```typst
#let phase(x) = {
  $underline(#box(
    inset: (bottom: -0.1658em, right: 0em),
    line(
      start: (0em, 1em),
      end: (0.4em, 0.2em),
      stroke: (thickness: 0.048em, cap: "round"),
    )
  )#x)$
}

$3phase(35degree)$
```
![image](https://github.com/typst/typst/assets/105704427/a77238d9-449f-4be8-91c4-f63cdbeff463)

**Comment by Dherse** [comment]
> I'm not sure this is common enough to be a built-in feature.

I disagree, in EE it's a very common notation especially (afaik) in French-spoken education context. That's how it was taught at my University and others I know of. I think it should be built in, it wouldn't be that much work to implement it just the way you described (with the line lining up a bit better).

**Comment by laurmaedje** [comment]
Isn't there a lot of field-specific notation that's used a lot and it still can't all be in the core language?

**Comment by Dherse** [comment]
Yes indeed, perhaps I am a bit biased because of my background, but I find this one to make a lot of sense...

**Comment by Enivex** [comment]
I am with Laurmaedje here. This makes a lot more sense as part of a field-specific package, and not built-in.
